src/parser.js

```diff
@@ -1539,6 +1573,12 @@ export class Parser extends Tokenizer {
         return this.markLocation(new Shift.ShorthandProperty(new Shift.Identifier(key.value)), startTokenIndex);
       }
     }
+    if (this.match(TokenType.LBRACK)) {
```

Comment:
we can, but I need to understand why it is necessary to test the following before going back into this.parseObjectPropertyKey().  Couldn't we hit eof() at any point inside parseObjectPropertyKey?

 if (this.eof() || token.type.klass == TokenClass.Punctuator) {